pkg/arvo/sur/btc-node-store.hoon

```diff
@@ -0,0 +1,180 @@
+          unlocked-until=(unit @ud)
+        ::
+          pay-tx-fee=@t
+        ::
+          hd-seed-id=(unit @ux)
+        ::
+          private-keys-enabled=?
+      ==
+    ::
+    +$  wallet
+      [name=@t attrs=(unit wallet-attr)]
+    ::
+    +$  wallets    (map @t wallet)
+    ::
+    +$  addresses  (list address)
+    --
+|%
+::
++$  btc-node-store-action   action:btc-rpc
++$  btc-node-store-update   update:btc-rpc
++$  btc-node-store-command  command:btc-rpc
+::
+++  btc-rpc
+  |%
+  +$  action
+    $%  :: [%abandon-transaction]
+        :: [%abort-rescan]
+        [%add-multisig-address ~]
+        :: [%backup-wallet ~]
+        :: [%bump-fee]
+        ::   Adds a new wallet to the list
+        ::
+        [%add-wallet name=@t warning=(unit @t)]
+        :: [%dump-privkey ~]
+        :: [%dump-wallet ~]
+        :: [%encrypt-wallet ~]
+        :: [%get-addresses-by-label]
+        :: [%get-address-info]
+        :: [%get-balance]
+        :: [%get-new-address]
+        :: [%get-raw-change-address]
+        :: [%get-received-by-address]
+        :: [%get-received-by-label]
+        :: [%get-transaction]
+        :: [%get-unconfirmed-balance]
+        :: [%get-wallet-info ~] -> This is replaced by %update
+        [%update-wallet name=@t attrs=wallet-attr]
+        [%import-address ~]
+        [%import-multi ~]
+        [%import-privkey ~]
+        [%import-pruned-funds ~]
+        [%import-pubkey ~]
+        [%import-wallet ~]
+        [%key-pool-refill ~]
+        :: [%list-address-groupings]
+        :: [%list-labels]
+        :: [%list-lock-unspent]
+        :: [%list-received-by-address]
+        :: [%list-received-by-label]
+        :: [%lists-in-ceblock]
+        :: [%list-transactions]
+        :: [%list-unspent]
+        :: [%list-wallet-dir ~]
+        [%list-wallets ~]
+        :: [%load-wallet ~]
+        :: [%lock-unspent]
+        :: [%remove-pruned-funds ~]
+        :: [%rescan-blockchain ~]
+        :: [%send-many ~]
+        :: [%send-to-address ~]
+        :: [%set-hd-seed ~]
+        :: [%set-label ~]
+        :: [%set-tx-fee ~]
+        :: [%sign-message ~]
+        :: [%sign-raw-transaction-with-wallet ~]
+        :: [%unload-wallet ~]
+        :: [%wallet-create-fundedpsbt ~]
+        :: [%wallet-lock ~]
+        :: [%wallet-passphrase ~]
+        :: [%wallet-passphrase-change ~]
+        :: [%wallet-process-psbt ~]
+        :: [%get-zmq-notifications]
+    ==
+  ::
+  +$  update
+    $%  ::  Wallet name has changed
+        ::
+        [%wallet-name name=@t]
+        ::  Updates wallet attributes
+        ::  FIXME: all attrs might need to be units...
+        ::
+        [%wallet-attrs name=@t attr=wallet-attr]
+    ==
+  ::
+  +$  command
+    $%  ::  Updated the default wallet stored in btc-node-store
+        ::
+        [%switch-wallet @t]
+      ::
+        ::  Loads an external wallet
+        ::
+        [%load-wallet @t]
+      ::
+        ::  TODO: do the actual syncing
+        ::
+        [%sync ~]
```

Comment:
I think that we should follow Bcoin's structure and segregate the node and wallet logic, so it might be better to postpone that so we have a clear idea of what goes in the wallet %hook/%store apps.